### PR TITLE
Add 'load chart from bookmark' functionality

### PR DESCRIPTION
**Pull request by RileyChang**
Clicking the 'select' icon on a bookmarked chart in the bookmarks list causes it to be loaded as the main viz. 

Fix https://github.com/uwdata/voyager2/issues/94

### CODE REVIEW

**Comment by RileyChang** [comment]
@kanitw 

Still TODO:
- Add new log type, as suggested in https://github.com/uwdata/voyager2/issues/94
- Remove old method of selecting a chart?
  - [x] selectAction
- [x] make `postSelectAction` work (scroll to the top) 
- [ ] even if you select in the bookmark list, how to still scroll to the top in the main view

File: src/components/vlplotgroup/vlplotgroup.js [hunk 1]
```diff
@@ -153,6 +153,17 @@ angular.module('vlui')
           return false;
         };
 
+        scope.select = function(chart) {
+          Logger.logInteraction(Logger.actions.SPEC_SELECT, chart.shorthand, {
+            list: scope.listTitle
+          });
+          Pills.parse(chart.vlSpec);
+          if (scope.postSelectAction) {
```
Comment (kanitw) [inline src/components/vlplotgroup/vlplotgroup.js]: Where is this from?   

We don't have `scope.postSelectAction` here.  

Thus, the main view won't automatically scroll to the top when you selects a plot in the main view. 
(But we should make it scroll to the top if we select a view in the bookmark view too.)

File: src/components/bookmarklist/bookmarklist.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ angular.module('vlui')
       restrict: 'E',
       replace: true,
       scope: {
-        highlighted: '=' // This one is really two-way binding.
+        highlighted: '=', // This one is really two-way binding.
+        postSelectAction: '='
```
Comment (kanitw) [inline src/components/bookmarklist/bookmarklist.js]: Does '<' work?

File: src/components/vlplotgroup/vlplotgroup.js [hunk 3]
```diff
@@ -163,6 +163,17 @@ angular.module('vlui')
...
 
+        scope.select = function(chart) {
+          Logger.logInteraction(Logger.actions.SPEC_SELECT, chart.shorthand, {
+            list: scope.listTitle
+          });
+          Pills.parse(chart.vlSpec);
+          if (scope.$parent.postSelectAction) {
+            scope.$parent.postSelectAction();
+          }
+          Modals.close('bookmark-list');
```
Comment (kanitw) [inline src/components/vlplotgroup/vlplotgroup.js]: Add  `//HACK` comment. This is okay for normal plots because bookmark isn't open yet.

**Pull request merged**